Top level `break` instruction causes panic on REPL

The following Goby code from https://github.com/goby-lang/goby/blob/master/vm/statement_test.go#L13 causes panic only on REPL:

```ruby
x = [1, 2, 3]
y = 0
x.each do |i|
  y += i
  if i == 2
    break
  end
end
y
```

```
» y
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x68 pc=0x43ef404]

goroutine 1 [running]:
github.com/goby-lang/goby/vm.(*VM).REPLExec(0xc4200a6780, 0xc4201e69c0, 0x5, 0x8)
	/Users/hachi8833/deve/golang/gopath/sys/src/github.com/goby-lang/goby/vm/repl.go:38 +0x4d4
github.com/goby-lang/goby/igb.StartIgb(0x4577c2d, 0x5)
	/Users/hachi8833/deve/golang/gopath/sys/src/github.com/goby-lang/goby/igb/repl.go:231 +0xd23
main.main()
	/Users/hachi8833/deve/golang/gopath/sys/src/github.com/goby-lang/goby/goby.go:29 +0x954
```

Note that the code works fine when running as a file.

And the following works fine on REPL; looks like the bug only occurs on top level `break` instruction.

```ruby
» def foo
¤   x = [1, 2, 3]
¤   y = 0
¤   x.each do |i|
¤     y += i
¤     if i == 2
¤       break
¤     end
¤   end
¤   y
» end
#»
» foo
#» 3
```

FYI: complied with Go 1.9.3.

Was looking into this today. Seem like something with the `break` statement.

```ruby
Goby 0.1.7 🙃 😚 😺
» [1, 2, 3].each do |i|
¤   break
» end
#» [1, 2, 3]
»
» y = 0
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x60 pc=0x44006b5]

goroutine 1 [running]:
github.com/goby-lang/goby/vm.(*VM).REPLExec(0xc4200a6820, 0xc4201fa6c0, 0x3, 0x4)
	/Users/julio/lib/go/src/github.com/goby-lang/goby/vm/repl.go:38 +0x435
github.com/goby-lang/goby/igb.StartIgb(0x459e471, 0x5)
	/Users/julio/lib/go/src/github.com/goby-lang/goby/igb/repl.go:231 +0xc13
main.main()
	/Users/julio/lib/go/src/github.com/goby-lang/goby/goby.go:29 +0x8fc
```

@hachi8833 

Ran another test (see snippet below) and the problem seems to be related to the top level but with the break statement being on the top level.

```ruby
Goby 0.1.7 🤓 😐 🤧
» def foo
¤   [1, 2, 3].each do |i|
¤     break
¤   end
¤   y = 0
» end
#»
» foo
#» 0
» y = 0
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x60 pc=0x44006b5]

goroutine 1 [running]:
github.com/goby-lang/goby/vm.(*VM).REPLExec(0xc4200a68c0, 0xc420204cc0, 0x5, 0x8)
	/Users/julio/lib/go/src/github.com/goby-lang/goby/vm/repl.go:38 +0x435
github.com/goby-lang/goby/igb.StartIgb(0x459e471, 0x5)
	/Users/julio/lib/go/src/github.com/goby-lang/goby/igb/repl.go:231 +0xc13
main.main()
	/Users/julio/lib/go/src/github.com/goby-lang/goby/goby.go:29 +0x8fc
```

--- a/vm/thread.go
+++ b/vm/thread.go
@@ -152,7 +152,18 @@ func (t *Thread) evalCallFrame(cf callFrame) {
 		t.Stack.Push(&Pointer{Target: result})
 		//fmt.Println(t.callFrameStack.inspect())
 		//fmt.Println("-----------------------")
-		t.callFrameStack.pop()
+
+		// this check is to prevent popping out a call frame that should be kept
+		// usually the top call frame would be the `cf` here
+		// but in some rare cases it could be other call frame, and if we pop it here we'll have some troubles in the later execution
+		//
+		// one example of the problem would be https://github.com/goby-lang/goby/issues/584
+		// the cause of this issue is that the "break" instruction always pops 3 call frames
+		// (this is necessary, please read the comments inside `Break` instruction for more detail)
+		// and because we already popped the `cf` during the `Break` instruction, what we pop here would be the top-level call frame, which causes the program to crash
+		if t.callFrameStack.top() == cf {
+			t.callFrameStack.pop()
+		}
 	}
 
 	t.removeUselessBlockFrame(cf)

--- a/vm/vm_test.go
+++ b/vm/vm_test.go
@@ -18,6 +18,24 @@ func TestVM_REPLExec(t *testing.T) {
 		{
 			[]string{
 				`
+x = [1, 2, 3]
+y = 0
+`,
+				`
+x.each do |i|
+  y += i
+  if i == 2
+    break
+  end
+end
+`,
+				`y`,
+			},
+			3,
+		},
+		{
+			[]string{
+				`
 				a, b = [3, 6]
 				a + b
 				`,

